### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by bqpd**

### CODE REVIEW

File: gpkit/nomials/math.py [hunk 1]
```diff
@@ -519,6 +523,8 @@ def sens_from_dual(self, la, nu, result):  # pylint: disable=unused-argument
         self.relax_sensitivity = la
         if self.sgp_parent:
             self.sgp_parent.relax_sensitivity = la
+            if getattr(self.sgp_parent, "sgp_parent", None):
+                self.sgp_parent.sgp_parent.relax_sensitivity = la
```
Comment (1ozturkbe) [inline gpkit/nomials/math.py]: Why are there two layers of sgp_parent here?

File: gpkit/solution_array.py [hunk 2]
```diff
@@ -70,9 +70,14 @@ def tight_table(self, _, ntightconstrs=5, tight_senss=1e-2, **kwargs):
...
     title = "Tightest Constraints"
-    data = [((-float("%+6.2g" % c.relax_sensitivity), str(c)),
-             "%+6.2g" % c.relax_sensitivity, id(c), c)
-            for c in self.model.flat() if c.relax_sensitivity >= tight_senss]
+    try:
+        data = [((-float("%+6.2g" % c.relax_sensitivity), str(c)),
+                 "%+6.2g" % c.relax_sensitivity, id(c), c)
+                for c in self.model.flat()
+                if c.relax_sensitivity >= tight_senss]
+    except AttributeError:
```
Comment (1ozturkbe) [inline gpkit/solution_array.py]: Agree that this is a better error message!

File: gpkit/nomials/math.py (hunk 1 above)
Comment (1ozturkbe) [reply gpkit/nomials/math.py]: Oh let me guess, because you can have relaxations and SP approximations?

Comment (bqpd) [reply gpkit/nomials/math.py]: yup.

**Comment by bqpd** [comment]
test models please

**Pull request merged**